servers/src/grin/sync/state_sync.rs

```diff
@@ -150,7 +150,7 @@ impl StateSync {
 				.chain
 				.get_block_header(&header_head.prev_block_h)
 				.unwrap();
-			for _ in 0..(horizon - horizon / 10) {
+			for _ in 0..(threshold - threshold / 10) {
```

Comment:
If horizon is `10,080` and threshold is `2,880` then _this_ is already the "percentage" - we are using approx 25% of the `horizon` by using the lower `threshold` value (we don't need to use 90% of threshold, we can just use 100% of it).

Reply:
@antiochp   this percentage (`90%` or `95%` as suggested by @hashmap ) seems a different thing from 2,880/10,080. 
As the comments line said `this still leaves time for download and validation to happen and stay within`.

If remove it and just use 100% here, we will always enter a txhashset download because of https://github.com/mimblewimble/grin/blob/42aa7a5d67eea3c7e7d085dd5e98fe918bea10a5/servers/src/grin/sync/state_sync.rs#L66-L72

Could you please think it again? I have no idea how to use 100% here and ensure state_sync.rs#L66-L72 get `need_state_sync` as false, after txhashset download and validation completed.

Reply:
Ah ok I see what you're saying. We transition to sync mode based on _either_ `tail` or the `sync_threshold` -

```
cmp::max(
	head.height.saturating_sub(tail.height),
	global::state_sync_threshold() as u64,
),
```

So a question - do we even need a `sync_threshold` if we now track the local `tail`?
If we know our local oldest full block (based on the last time we pruned/compacted the db or the last time we sync'd) then is that sufficient?

What if we drove the `need_state_sync` purely off our oldest full block (the `tail`)`?